### DISCUSSION TITLE
Kafka Producer : Golang or Quarkus ?

### DISCUSSION THREAD

**Discussion by ares-b** [quarkusio/quarkus#30110] [Q&A]
Hello,

ATM I'm using an Akka Cluster running on Kubernetes to collect Data from APIs & Websockets in near real-time, I'm using the Akka Actor's capabilities to ensure fault tolerance and persistence of data on Cassandra.

But, Lightbend announced few months ago that Akka is no longer under Apache License, they moved to BSL license 🤦🏾‍♂️

I wanna move out from Akka, and was thinking about using Golang or Quarkus to collect my data into some Kafka topics before processing the data with Flink and writing to Cassandra.

I found [this benchmark on Quarkus vs Golang](https://gregscode.tech/native-battle-quarkus-vs-spring-vs-golang/), overall, it says that Quarkus does better than Golang, but I find the benchmark quite skewed, as it's only using one core and 1gb RAM with no scaling and no fault recovery mecanisms. So I thought I'd be better asking the community.

Have any one of you guys used both ? Which one do you think is best suitable for my use case ?

Thanks,

Ares

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
Well, expect the opinion here to be a bit biased. 

I think in your case, the migration would be simpler with Quarkus (at least you stay on the JVM so tuning/monitoring won't be that different).
